ast-nodes/if-statement.dart

```diff
@@ -12,6 +12,32 @@ class IfStatement implements Statement {
 
   // TODO: implement .parse()
 
+  factory IfStatement.parse(Iterable<Token> tokens) {
+    var iterator = tokens.iterator;
+    checkNext(iterator, RegExp('if\$'), "Expected 'if'");
+    var expressionBody = consumeUntil(iterator, RegExp("^then\$"));
+    checkNext(iterator, RegExp('then\$'), "Expected 'then'");
```

Comment:
This should be `checkThis` instead of `checkNext` cause `consumeUntil` leaves the iterator at the terminal

Reply:
line 17 should be checkNext no? you're talking about line 19

Reply:
No, line 17 should be `checkNext`, line 19 should be `checkThis`